routes/map.js

```diff
@@ -1,116 +1,217 @@
+      let nodes = []
+      let relationships = []
+      collection.insertOne({ userID: userID, mapID: randomUUID(), nodes: nodes, relationships: relationships})
+        .then(result => {
+          res.status(200).json(result)
+        })
+        .catch(err => {
+          res.status(400).json({error: "Could not create new map"})
+        })
+    })
+  }
 });
 
 /**
- * Get endpoint, which takes the userID parameter and returns the users map
+ * Get endpoint, expects mapID and returns all data in a map document, can be the raw JSON object.
  */
 
 router.get("/get", function (req, res, next) {
-  const { userID } = req.body;
-  let inDb = JSON.parse(fs.readFileSync(__dirname + "/../public/tempDB.json", 'utf-8'));
-  const found = inDb.find((inDBobj) => inDBobj.userID === userID);
+  // Expects mapID
+  const { mapID } = req.body;
 
-  if (found) {
-    const user = inDb.filter((inDBobj) => inDBobj.userID === userID);
-    res.status(200).json(user);
-  } else {
-    res.status(404).json({ message: "User not found" });
+  // Ensure mapID is specified
+  if (!mapID && mapID !== 0) {
+    res.status(400).send("Must specify mapID!");
+  }
+  else {
+    // Connect to the database
+    client.connect(() => {
+      // Navigate to the correct database and collection
+      let database = client.db("blurp")
+      let collection = database.collection("maps")
+
+      // Find map by mapID
+      collection.find({ mapID: mapID })
+        .toArray()
+        .then((ans) => {
+          if(ans.length === 0)
+            res.status(400).json({ error: "Could not find map"})
+          else
+            res.status(200).json(ans)
+        })
+    })
   }
 });
 
 /**
- * Update endpoint, which takes the relationshipID parameter and updates the relationship
- * NOTE: "update" for this data structure should have some of (but not necessarily all of)
- * nodeOne, nodeTwo, description, relationshipType
+ * Update endpoint, expects userID, newUserID, and mapID. Checks to see if user is trying to update a node/relationship
+ * array, resulting in a failure. Then checks if user owns the map. If user owns the map, then update userID with
+ * newUserID, else failure. 
  */
 
 router.patch("/update", function (req, res, next) {
-  const { relationshipID, update } = req.body;
-  let inDB = JSON.parse(fs.readFileSync(__dirname + "/../public/tempDB.json", 'utf-8'));
-  const found = inDB.findIndex((inDBobj) => inDBobj.relationshipID === relationshipID);
-
-  if (found !== -1) {
-    Object.assign(inDB[found], update);
-    fs.writeFile(__dirname + "/../public/tempDB.json", JSON.stringify(inDB), function (err) {
-        if (err) {
-          res.status(400).send("Issue writing file!");
-        }
-    });
-    res.status(200).json(inDB[found]);
-  } else {
-    res.status(404).json({ mess: "relationship not found" });
+  // Expect userID, newUserID, and mapID
+  const { userID, newUserID, mapID } = req.body
```

Comment:
The only thing I would change is that we will expect many different attributes as time goes on and we enhance the map behavior. The route may contain other things to change than the userID that owns the map - is it possible to make the expected body something like:
body = { 
  userid = 1,
  mapid = 1,
  updates = {
    userid = 2,
    <etc.>
  }
}